scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -848,6 +854,24 @@ def c2(x):
         assert_(np.all(res.x >= np.array(bounds)[:, 0]))
         assert_(np.all(res.x <= np.array(bounds)[:, 1]))
 
+        # now repeat the same solve, using the same overall constraints,
+        # but using a sparse matrix for the LinearConstraint instead of an
+        # array
+
+        L = LinearConstraint(dok_matrix(A), -np.inf, b)
+
+        # using a lower popsize to speed the test up
+        res = differential_evolution(f, bounds, strategy='best1bin', seed=1234,
+                                     constraints=(L), popsize=2)
```

Comment:
It's not exactly the same set of constraints as the previous example. There the constraints are `(L, N, L2, N2)`. Which ones did you intend to use? If you want the same constraints shouldn't you use:
```
        L = LinearConstraint(dok_matrix(A[:5, :), -np.inf, b[:5])
        L2 = LinearConstraint(A[5:6, :], -np.inf, b[5:6])
        N = NonlinearConstraint(c1, -np.inf, b[6:8])
        N2 = NonlinearConstraint(c2, -np.inf, b[8:9])
        constraints = (L, N, L2, N2)
        with suppress_warnings() as sup:
            sup.filter(UserWarning)
            res = differential_evolution(f, bounds, strategy='rand1bin',
                                         seed=1234, constraints=constraints,
                                         popsize=2)
```

Reply:
I moved the test above that one, between the original 1st one and the original 2nd one, and it's exactly the same as the 1st.